### DISCUSSION TITLE
Chatbot on top of docs

### DISCUSSION THREAD

**Discussion by viraj-lunani** [brexhq/substation#210] [Ideas]
Hey Team,

We've been starting to increase our dependence on substation and writing more configs. We noticed a few things about the documentation that makes the process of writing configs a little tedious and not the best dev experience.

1. I think having more and better examples for all of the different transforms would really help us figure out how to use them. It feels like there is a lot of trial and error and writing configs is just a headache that no one on the team looks forward to.

2. I feel like this is a perfect use case for a chatbot on top of the docs. E.g. "I am trying to write X transform, how can I do this with substation". I haven't looked into this too much but I'm sure there a lot of solutions for this for readme docs. Saw [this](https://docs.readme.com/main/docs/owlbot) with a _very_ brief google search (not sure if that specifically would work but sure there's something like it).

Anyways just some feedback to make the dev experience for writing configs better. Overall we've been having a great experience with substation and the team is super helpful as always!

**Comment by jshlbrd** [comment]
@viraj-lunani Thanks for the message, I'll respond in line below. 👍 

> I think having more and better examples for all of the different transforms would really help us figure out how to use them. It feels like there is a lot of trial and error and writing configs is just a headache that no one on the team looks forward to.

Sorry to hear that the team doesn't like writing configs, there's room for improvement but in my experience this is roughly equivalent to what it's like managing configurations in similar systems like Logstash and Fluentd. If you have specific ideas on how to make the project easier to use, then let us know.

Also, the project has [configuration examples](https://github.com/brexhq/substation/tree/main/examples/config) and we accept pull requests, so feel free to add your own examples -- I'm sure others would appreciate it!

> I feel like this is a perfect use case for a chatbot on top of the docs. E.g. "I am trying to write X transform, how can I do this with substation". I haven't looked into this too much but I'm sure there a lot of solutions for this for readme docs. Saw [this](https://docs.readme.com/main/docs/owlbot) with a very brief google search (not sure if that specifically would work but sure there's something like it).

Substation is supported by security engineers at Brex in our spare time and the Readme Owlbot AI costs money -- we won't be adding support for anything like that (and we are unlikely to support anything that does not directly contribute to the features, reliability, or performance of the project). 

In my experience tools like GitHub Copilot are great "pair programmers" for writing Substation configurations, so I'd encourage you to try a tool like that.
